server/src/test/java/org/elasticsearch/transport/InboundDecoderTests.java

```diff
@@ -55,18 +54,34 @@ public void testDecode() throws IOException {
         } else {
             threadContext.addResponseHeader(headerKey, headerValue);
         }
-        OutboundMessage message;
-        final Writeable body;
-        if (isRequest) {
-            body = new TestRequest(randomAlphaOfLength(100));
-            message = new OutboundMessage.Request(threadContext, body, TransportVersion.current(), action, requestId, false, null);
-        } else {
-            body = new TestResponse(randomAlphaOfLength(100));
-            message = new OutboundMessage.Response(threadContext, body, TransportVersion.current(), requestId, false, null);
-        }
 
         try (RecyclerBytesStreamOutput os = new RecyclerBytesStreamOutput(recycler)) {
-            final BytesReference totalBytes = message.serialize(body, os);
+            final BytesReference totalBytes;
+            if (isRequest) {
+                totalBytes = OutboundHandler.serialize(
```

Comment:
I guess we could add some overloads to shorten this thing, but not sure it's worth it, it's not that many spots and it's test only code.

Reply:
Yeah no big deal IMO